compiler/src/iree/compiler/Codegen/LLVMCPU/Utils.cpp

```diff
@@ -58,4 +58,44 @@ bool hasI8mmFeature(IREE::HAL::ExecutableTargetAttr targetAttr) {
   return hasFeature(targetAttr, "+i8mm");
 }
 
+bool transposeLoweringPreconditionAArch64SME(linalg::GenericOp genericOp) {
```

Comment:
I think calling this `transposeLoweringPreconditionAArch64SME` makes this look more specific to ArmSME than it really is. This is just something like `isLinalgGeneric2DTransposition`.